### PR TITLE
Option to disable Rspamd local networks

### PR DESCRIPTION
**Pull request by Zegorax**
To give more context about this PR, here is a copy paste of the comment I've posted on issue #192.

I spent a few hours today trying to dig the issue and I finally understood why it was not working.

### Root cause
As defined in https://github.com/anonaddy/anonaddy/blob/ad77a6ccf90a974408370df9e574356ff2a04fef/app/Console/Commands/ReceiveEmail.php#L156 , AnonAddy checks whether the headers `X-AnonAddy-Spam` or `X-AnonAddy-Dmarc-Allow` are present or not.
Those headers are added ONLY if Rspamd is enabled and correctly configured to inject those headers, which is not the case here because the environment variable `RSPAMD_ENABLE` is set to `false` by default.

### Second problem
Even after enabled Rspamd, the messages would still not get delivered. I then explored the Rspamd dashboard to understand why. After seeing the messages, they would not have the DMARC_POLICY_ALLOW tag attached to them. As per the default Rspamd config in this repo, the header `X-AnonAddy-Dmarc-Allow` will only get injected if DMARC_POLICY_ALLOW is added.

After setting the Rspamd logs to a more verbose option, I figured out that since I'm running my mailserver also in containers, they are (obviously) in a local network which Rspamd does not like by default. 
Because, by default, Rspamd will skip DKIM, DMARC and SPF checks if a new message comes from a local network. This behavior is defined in `/etc/rspamd/options.inc`. I therefore removed the content of the `local_addrs` variable defined there.

I tried again, and eureka, it works! So, TL;DR:

### Solution
Update your docker-compose as the following:
```
entrypoint: bash -c "sed -i 's/local_addrs.*$$/local_addrs=[]/' /etc/rspamd/options.inc && /init"

...
environment:
    - "RSPAMD_ENABLE=true"
```

This PR introduces the fix for the local_addrs variable containing the Docker internal networks, it would therefore prevent Rspamd from analyzing the requests and it will not inject the X-AnonAddy-... headers.

### CODE REVIEW

**Comment by Zegorax** [comment]
@crazy-max Could you review this PR  and merge it if approved ?

**Comment by willbrowningme** [comment]
Has anyone else tested this and confirmed it is working? I think we also need to make Rspamd required instead of optional too, perhaps with a strong random default web password `RSPAMD_WEB_PASSWORD` if none is provided.

File: rootfs/etc/cont-init.d/60-svc-rspamd.sh [hunk 1]
```diff
@@ -28,3 +28,7 @@ with-contenv
 /usr/sbin/rspamd -i -f -u rspamd -g rspamd
 EOL
 chmod +x /etc/services.d/rspamd/run
+
+# Empty the local_addrs array to avoid having Rspamd skip DMARC and SPF checks if the mailserver is running in a container as well (therefore being in a local network).
+# Required since AnonAddy checks the headers injected by Rspamd. See https://github.com/anonaddy/docker/issues/192#issuecomment-1518111988
+sed -i 's/local_addrs.*$/local_addrs=[]/' /etc/rspamd/options.inc
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/60-svc-rspamd.sh:34]: Sorry for the delay, can we make this configurable through an env var? Maybe smth like `RSPAMD_NO_LOCAL_ADDRS` which defaults to `false`.

Comment (crazy-max) [reply rootfs/etc/cont-init.d/60-svc-rspamd.sh]: I have gone ahead and added an env var

**Pull request merged**